bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/SonosXMLParser.java

```diff
@@ -127,18 +128,18 @@ public static List<SonosEntry> getEntriesFromString(String xml) {
      * @param xml
      * @return The value of the desc xml tag
      * @throws SAXException
+     * @throws ParserConfigurationException
      */
-    public static @Nullable SonosResourceMetaData getResourceMetaData(String xml) throws SAXException {
-        XMLReader reader = XMLReaderFactory.createXMLReader();
-        reader.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+    public static @Nullable SonosResourceMetaData getResourceMetaData(String xml)
+            throws SAXException, ParserConfigurationException {
```

Comment:
I am not sure to understand why this one is different and you throw `ParserConfigurationException` rather than catching it.

Reply:
I think the only reason was to not change anything not strictly needed to not extend scope more than needed.

Previously `SAXException` could be thrown, and the caller would handle that. Now `ParserConfigurationException` is added to the list, and the caller will handle that as well. For some of the other methods in this class, `SAXException` is caught and handled within the method itself, and in these cases, the same is now the case with `ParserConfigurationException`.

Reply:
To be more specific, some lines of this method outside the try/catch could throw `SAXException`. This would then be caught by `endElement` and ignored/logged there. Now, with the new API `ParserConfigurationException` can also be thrown. Therefore, this exception is now also caught by `endElement`. This way the flow should be the same, i.e. this part of the refactoring is kept strictly to adapt to the new API.

Some of the other methods handle all exceptions internally, so they look at bit different.